wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt

```diff
@@ -207,6 +209,81 @@ class EditProfileActivity : LockScreenActivity() {
             imitateUserInteraction()
             cropProfilePicture()
         }
+
+        takePicturePermissionLauncher =  registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+            if (isGranted) {
+                takePicture()
+            } else {
+                // Handle the case where permission is denied
+            }
+        }
+
+        takePictureLauncher = registerForActivityResult(ActivityResultContracts.TakePicture()) { isSuccess: Boolean ->
+            if (isSuccess) {
+                editProfileViewModel.onTmpPictureReadyForEditEvent.postValue(editProfileViewModel.tmpPictureFile)
+            }
+            turnOnAutoLogout()
+        }
```

Comment:
This refactoring also solved the problem of failure of the camera on Android 12 and 14 emulators, though the old code did work on a real Android 13 device.